docs/getting_started.rst

```diff
@@ -1,135 +1,288 @@
-a :class:`kartothek.core.dataset.DatasetMetadata` object is returned. This is the main
-class holding all information about the parts and schema of the dataset.
 
-The most interesting ones for this guide are ``tables`` and ``partitions``.
-Each dataset can have multiple tables, each table is a collection of files
-that all have the same schema. These files are called ``partitions`` in
-``kartothek``.
+.. admonition:: Storage backends
 
-As we neither have explicitly defined the name of the table nor the name
-of the created partition, ``kartothek`` has used the default table name
-``table`` and used a generated UUID for the partition name.
+    The import path of this function already gives us a hint about the general
+    structuring of the ``kartothek`` modules. In :mod:`kartothek.io` we have all
+    the building blocks to build data pipelines that read and write from/to storages.
+    The next module level (e.g. ``eager``) describes the scheduling backend.
+
+    The scheduling backends `currently supported` by kartothek are:
+
+    - ``eager`` runs all execution immediately and on the local machine.
+    - ``iter`` executes operations on the dataset on a per-partition basis.
+      The standard format to read/store dataframes in ``iter`` is by providing
+      a generator of dataframes.
+    - ``dask`` is suitable for larger datasets. It can be used to work on datasets in
+      parallel or even in a cluster by using ``dask.distributed`` as the backend.
+      There are also ``dask.bag`` and ``dask.dataframe`` which support I/O operations
+      for the respective `dask`_ objects.
+
+
+After calling :func:`~kartothek.io.eager.store_dataframes_as_dataset`,
+a :class:`~kartothek.core.dataset.DatasetMetadata` object is returned.
+This class holds information about the structure and schema of the dataset.
 
 .. ipython:: python
 
-   dm.tables
-   dm.partitions
+    dm.tables
+    sorted(dm.partitions.keys())
+
+
+For this guide, two attributes that are noteworthy are ``tables`` and ``partitions``:
+
+- Each dataset has one or more ``tables``, where each table is a logical collection of data,
+  bound together by a common schema.
+- ``partitions`` are the physical "pieces" of data which together constitute the
+  contents of a dataset. Data is written to storage on a per-partition basis.
+  See the section on partitioning for further details: :ref:`partitioning_section`.
 
 For each table, ``kartothek`` also tracks the schema of the columns.
-When not specified explicitly on write, it is inferred from the passed data.
+Unless specified explicitly on write, it is inferred from the passed data.
 On writing additional data to a dataset, we will also check that the schema
-of the new data matches the schema of the existing data. If it doesn't, we will
-raise an exception.
+of the new data matches the schema of the existing data.
+
+While two tables may have different schemas, every file belonging to a given table needs
```

Comment:
This paragraph is too technical and prob too much information for the getting started guide. This is/should be covered in the writeup of the storage specification but not the getting started guide https://kartothek.readthedocs.io/en/latest/spec/index.html